tests/parser/persistentPageIdentifierFunction.txt

```diff
@@ -0,0 +1,45 @@
+!! options
+parsoid-compatible
+version=2
+!! end
+
+# Force the test runner to ensure the extension is loaded
+!! functionhooks
+ppid
+!! endfunctionhooks
+
+!! test
+Returns nothing without a page context
+!! wikitext
+{{#ppid:}}
+!! html
+!! end
+
+!! article
+Test Page
+!! text
+!! endarticle
+
+!! test
+Returns persistent identifier for page
+!! options
+title=[[Test Page]]
+!! wikitext
+{{#ppid:}}
+!! html
+<p>id-8
```

Comment:
How come it is 8? Does the number depend on if you run the unit tests first? I was going to merge despite the internal :scream: , but now I'm actually concerned.

Do we lose something if we use a StubIdGenerator?

Reply:
![image](https://github.com/user-attachments/assets/263f9cbc-87c2-454c-8293-5f9e10bdc20c)
For me `id-8` fails but `id=9` passes.

Reply:
A bunch of default files get created here (`$localRepo->newFile`):
7 in MW 1.39-1.42: https://github.com/wikimedia/mediawiki/blob/REL1_39/tests/parser/ParserTestRunner.php
8 in MW 1.43: https://github.com/wikimedia/mediawiki/blob/REL1_43/tests/parser/ParserTestRunner.php